run.sh

```diff
@@ -4,20 +4,30 @@ set -e
 
 neo_image_name=${NEO_IMAGE_NAME:-mlnx-neo}
 neo_container_name=${NEO_CONTAINER_NAME:-mlnx_neo}
+neo_volume_name=${NEO_CONTAINER_NAME:-mlnx_neo_data}
 neo_startup_config_path=${NEO_STARTUP_CONFIG_PATH:-}
 
 extra_volumes=""
-if [[ -n $neo_startup_config_path ]] then
+if [[ -n $neo_startup_config_path ]]; then
    extra_volumes=" -v $neo_startup_config_path:/usr/bin/mlnx-neo-configure"
 fi
 
+# Volume for persistent NEO data
+if [[ -z $(docker volume ls --filter name=$neo_volume_name -q) ]]; then
+    docker volume create $neo_volume_name
+fi
+
 # Start the NEO container.
 docker run \
     --detach \
     --name $neo_container_name \
-    --network host \
-    --privileged \
+    -p 443:443 \
+    -p 162:162/tcp \
+    -p 162:162/udp \
+    -p 7654:7654 \
+    --tmpfs /run \
     -v /dev/log:/dev/log \
     -v /sys/fs/cgroup:/sys/fs/cgroup:ro \
```

Comment:
We actually normally use an ansible role to deploy this container: https://github.com/stackhpc/ansible-role-mlnx-neo. You can see the container config, which is a little different to this: https://github.com/stackhpc/ansible-role-mlnx-neo/blob/5f5606ebe2e592d6b4db50859f23b5e02bffea9a/defaults/main.yml#L13

In particular, we don't specify /dev/log:/dev/log or /sys/fs/cgroup:/sys/fs/cgroup:ro. It's possible these were a hangover from the similar docker-mlnx-ufm container.

Reply:
This is what you will see if you try start it without cgroup and tmpsfs (unless you include --privileged switch):
```bash
$ docker run -it mlnx-neo
Failed to mount tmpfs at /run: Operation not permitted
Failed to mount cgroup at /sys/fs/cgroup/systemd: Operation not permitted
[!!!!!!] Failed to mount API filesystems, freezing.
```

Will drop /dev/log from it.